patches/minecraft/net/minecraft/item/HoeItem.java.patch

```diff
@@ -12,9 +12,26 @@
        World world = p_195939_1_.func_195991_k();
        BlockPos blockpos = p_195939_1_.func_195995_a();
 -      if (p_195939_1_.func_196000_l() != Direction.DOWN && world.func_180495_p(blockpos.func_177984_a()).func_196958_f()) {
+-         BlockState blockstate = field_195973_b.get(world.func_180495_p(blockpos).func_177230_c());
 +      int hook = net.minecraftforge.event.ForgeEventFactory.onHoeUse(p_195939_1_);
 +      if (hook != 0) return hook > 0 ? ActionResultType.SUCCESS : ActionResultType.FAIL;
 +      if (p_195939_1_.func_196000_l() != Direction.DOWN && world.func_175623_d(blockpos.func_177984_a())) {
-          BlockState blockstate = field_195973_b.get(world.func_180495_p(blockpos).func_177230_c());
++         BlockState blockstate = getHoeTillingState(world.func_180495_p(blockpos), world, blockpos, p_195939_1_.func_195999_j(), p_195939_1_.func_195996_i());
           if (blockstate != null) {
              PlayerEntity playerentity = p_195939_1_.func_195999_j();
+             world.func_184133_a(playerentity, blockpos, SoundEvents.field_187693_cj, SoundCategory.BLOCKS, 1.0F, 1.0F);
+@@ -47,4 +49,14 @@
+ 
+       return ActionResultType.PASS;
+    }
++   
```

Comment:
There is random white space being added to this line, it should just be a blank line.